### PR TITLE
Sphinx-build docs and background work for RTD

### PR DESCRIPTION
**Pull request by duytnguyendtn**
This PR gets the docs to properly build via sphinx and does some background work to begin hooking the project up to RTD

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+version: 2
+
+build:
+  os: ubuntu-20.04
+  apt_packages:
+    - graphviz
+  tools:
+    python: "3.9"
```
Comment (rosteen) [inline .readthedocs.yaml:8]: Might as well use 3.10 or 3.11 to start out, get ahead of future deprecations.

Comment (duytnguyendtn) [reply .readthedocs.yaml]: Good idea! Done in https://github.com/spacetelescope/slitlessutils/pull/25/commits/037317a9904b5afe2ce3b13cc4d9a2301678404f

**Comment by rosteen** [comment]
Does RTD also need the `Makefile` to actually run the sphinx commands? In Jdaviz I always build the docs locally with `make html`...not sure if that's a handy extra or actually needed for RTD.

**Comment by duytnguyendtn** [comment]
This is something I'd like to explore to see if it's actually necessary for RTD, rather than blindly add one. As per https://www.sphinx-doc.org/en/master/usage/quickstart.html#running-the-build, it would suggest it may be a convenience feature. If not, I'll add a followup PR to add it

**Comment by rosteen** [comment]
Any reason not to finish setting up RTD as part of this PR so we can see if it works before merging this?

**Comment by duytnguyendtn** [comment]
> finish setting up RTD

Sorry, are there other steps I'm missing? I figured I needed the `.readthedocs.yaml` in the repo first before I can setup the RTD hooks on that side?

**Comment by duytnguyendtn** [comment]
Reading up on https://docs.readthedocs.io/en/stable/config-file/v2.html, my impression was that the `.readthedocs.yaml` file was required to setup the RTD hooks properly, hence my desire to get this PR in first. It's been quite a while since I've setup RTD, so let me know if there's a different process!

**Review by rosteen — APPROVED** [review]
Got it, I wasn't sure. Let's get this in then, anything else needed can be a follow-up.

**Pull request merged**